packages/program/src/functions/base-invocation-scope.ts

```diff
@@ -410,12 +403,24 @@ export class BaseInvocationScope<TReturn = any> {
     const gasLimitSpecified = !!this.txParameters?.gasLimit || this.hasCallParamsGasLimit;
     const gasPriceSpecified = !!this.txParameters?.gasPrice;
 
+    const { gasLimit, gasPrice } = transactionRequest;
+
     if (!gasLimitSpecified) {
       transactionRequest.gasLimit = gasUsed;
+    } else if (gasLimit.lt(gasUsed)) {
+      throw new FuelError(
+        ErrorCode.GAS_LIMIT_TOO_LOW,
+        `Gas limit '${gasLimit}' is lower than the required: '${gasUsed}'.`
+      );
```

Comment:
I suppose it'd create more problems than solutions if we tried reusing the `validateGas` helper, right?

https://github.com/FuelLabs/fuels-ts/blob/572e85164b5c98843541e0015f077b4f11a7ba9b/packages/account/src/account.ts#L529-L534

Reply:
@arboleya Yeah, I've had the same though. 

We could create a helper function to validate and throw this error and place it inside the `errors` package. It makes more sense to exist there since this error is being thrown in more than one package. 

But in the end, I thought this idea was overengineering.